nt-list/src/list/base.rs

```diff
@@ -457,17 +459,22 @@ where
         }
     }
 
-    pub(crate) fn containing_record(&self) -> &E {
-        unsafe { &*self.element_ptr() }
+    pub(crate) unsafe fn containing_record<'a>(ptr: *const Self) -> &'a E {
+        unsafe { &*Self::element_ptr(ptr) }
     }
 
-    pub(crate) fn containing_record_mut(&mut self) -> &mut E {
-        unsafe { &mut *(self.element_ptr() as *mut E) }
+    pub(crate) unsafe fn containing_record_mut<'a>(ptr: *mut Self) -> &'a mut E {
+        unsafe { &mut *Self::element_ptr_mut(ptr) }
     }
```

Comment:
`element_ptr()` has originally been written as a common function for `containing_record()` and `containing_record_mut()`.
Now that `containing_record()` uses `element_ptr()`, and `containing_record_mut()` uses `element_ptr_mut()`, and both are the only customers of the "element_ptr" functions, it makes sense to move it all to `containing_record()`/`containing_record_mut()` and get rid of `element_ptr()`/`element_ptr_mut()`.

Reply:
Inlining `element_ptr_{mut}()` also makes the implementation nicely parallel with the `Nt{Single}ListHead::entry` function, so sounds good to me either way.